polaris.shopify.com/src/components/Examples/Examples.tsx

```diff
@@ -15,26 +15,63 @@ interface Props {
   examples: [Example];
 }
 
+// https://stackoverflow.com/a/60338028
+function formatHTML(html: string): string {
+  const tab = "  ";
+  let result = "";
+  let indent = "";
+
+  html.split(/>\s*</).forEach((element) => {
+    if (element.match(/^\/\w/)) {
+      indent = indent.substring(tab.length);
+    }
+    result += indent + "<" + element + ">\r\n";
+
+    if (element.match(/^<?\w[^>]*[^\/]$/) && !element.startsWith("input")) {
+      indent += tab;
+    }
+  });
+
+  return result.substring(1, result.length - 3);
+}
+
 const Examples = (props: Props) => {
   const { examples } = props;
   const [selectedIndex, setSelectedIndex] = useState(0);
+  const [htmlCode, setHTMLCode] = useState("");
 
   const [iframeHeight, setIframeHeight] = useState("400px");
-  const iframeRef = useRef<HTMLIFrameElement>(null);
 
   const handleExampleLoad = () => {
-    const exampleElement =
-      iframeRef?.current?.contentWindow?.document?.getElementById(
-        "polaris-example"
-      );
-    const padding = 192;
-    let newHeight = padding;
-
-    if (exampleElement) {
-      newHeight += exampleElement?.offsetHeight;
-    }
+    let attempts = 0;
+
+    const waitForExampleContent = setInterval(() => {
```

Comment:
Just because the iframe's `onLoad` handler fires doesn't mean that React example has had the time to render. This timer polls the iframe to check if the example has _actually_ been rendered.